src/utils/NetworkGuard.ts

```diff
@@ -38,28 +28,56 @@ const validate = (() => {
   };
 })();
 
-function isMCQOptions(obj: unknown): obj is MCQOptions {
+/**
+ * Validates tags to ensure they meet the specified formats.
+ */
+function isTag(tag: string): boolean {
+  const isTaxonomy = tag.includes(":") && tag.split(":").length === 2;
+  const isSearchTag = !tag.includes(":") && !tag.includes(" ");
+  return isTaxonomy || isSearchTag;
+}
+
+function validateTags(arr: string[], checkAll: boolean = false): boolean {
+  if (!validate.isArray(arr, validate.isString)) {
+    return false;
+  }
+  return checkAll ? arr.every(isTag) : arr.some(isTag);
+}
+
+function isMCQOptions(obj: DataMCQOptions): boolean {
   return (
     validate.isObject(obj) &&
     validate.isString(obj.optionValue) &&
     (obj.optionCorrect === undefined || validate.isBoolean(obj.optionCorrect))
   );
 }
 
-function isMCQuestion(obj: unknown): obj is MCQuestion {
+function isMCQuestion(obj: DataMCQuestion): boolean {
   return (
     validate.isObject(obj) &&
```

Comment:
@freshc0w - We probably need to simplify this condition.

There is too much validate chaining going on here. Maybe this could be divided or it could be reduced?